### DISCUSSION TITLE
Units

### DISCUSSION THREAD

**Discussion by morteeke** [brittonhayes/aos#28] [General]
Hey @brittonhayes,

I hope I am not bothering you to much. 😅 

I was wondering if you already added some more units to the API. In case you did, I only get a response with 2 units.

**Comment by brittonhayes** [comment]
Hey @morteeke! No problem at all 😊 questions and issues are totally welcome and you've never gotta worry about asking me stuff!

To answer your question, it's kinda the same situation as why there's no armies in the /armies response - I've only added two units at the moment.

I wanna be honest with ya since you've been so kind and patient. 

The reason there isn't much data in the API is that I made this project for a couple friends who play Sigmar so they could make fun little tools within the AOS universe for their games together. I haven't personally played Warhammer but I've written a lot of code so I made this project as a friendly gesture for friends with the hope that they'd add the data since they know and love the game!

That plan unfortunately didn't pan out and that's got me in a spot where I've got an API that I'm proud of for a game I've never played and it's missing a TON of data.

Soooo,to summarize this, my hope is that folks like yourself will open pull requests to add the data you want to see in the api while I can focus on the codebase, features, maintenance, and deployments.

I would love to review and approve them. It just involves you and others adding lines to those fixtures/*.yaml files and the new data will automatically deploy and be in the API upon merge! I'm also more than happy to help in any way to make this API meet your needs 😊

**Reply by morteeke** [reply]
Hey @brittonhayes,

I was wondering if there is a way to get all units of a certain allegiance. I noticed there is a way to get units of a certain grand alliance (like order), but I would like to find units according to alegiance (like 'stormcast eternals'.

Thank you in advance!

Best regards,
Morteeke

**Reply by brittonhayes** [reply]
Well hey there friend!! 👋 after looking over the units service, looks like you're right! We've got pretty low support for filtering via query parameters. 

This should be super easy to add so I can take a look after work today, @morteeke 😊

**Reply by morteeke** [reply]
Whenever you have time!

Thanks! 😄

**Reply by brittonhayes** [reply]
@morteeke separate question, did you get around to adding all those units you wanted in the API in your fork? 

Would love to merge any added data into this repository if you have!😊

**Reply by morteeke** [reply]
@brittonhayes not yet, I'm sorry. The last months I have been to busy.

I will make work of it in the next few days.

**Reply by brittonhayes** [reply]
No need to apologize at all!! Was just curious 😊 I know how it is. Just wanted you to know your contributions are always welcome, no matter when.

**Reply by brittonhayes** [reply]
@morteeke The feature is in now as of #32 for both Graphql and REST! I've also cleaned up the list of allegiances in `/allegiances`
